code/modules/antagonists/demon/demons.dm

```diff
@@ -0,0 +1,172 @@
+	if(!chosen_sin)
+		return
+	if(chosen_sin in demonsins)
+		demonsin = chosen_sin
+	new_owner.add_antag_datum(src)
+	message_admins("[key_name_admin(admin)] has demonized [key_name_admin(new_owner)].")
+	log_admin("[key_name(admin)] has demonized [key_name(new_owner)].")
+
+/datum/antagonist/sinfuldemon/antag_listing_name()
+	return ..() + "(, demon of [demonsin])" // Boris Smith, demon of Wrath
+
+/datum/antagonist/sinfuldemon/greet()
+	to_chat(owner.current, span_warning("<b>You remember your link to the infernal. You are a demon of [demonsin] released from hell to spread sin amongst the living.</b>"))
+	to_chat(owner.current, span_warning("<b>However, your infernal form is not without weaknesses.</b>"))
+	to_chat(owner.current, "You are incredibly vulnerable to holy artifacts and influence.")
+	to_chat(owner.current, "While blessed with the unholy ability to transform into your true form, this form is extremely obvious and vulnerable to holy weapons.")
+	to_chat(owner.current, "[span_warning("Do your best to complete your objectives without unnessecary death, unless you are a wrathful demon.")]<br>")
+	owner.announce_objectives()
+	.=..()
+
+/datum/antagonist/sinfuldemon/on_gain()
+	forge_objectives()
+	owner.special_role = "sinfuldemon"
+	owner.current.faction += "hell"
+	if(owner.assigned_role == "Clown" && ishuman(owner.current))
+		var/mob/living/carbon/human/S = owner.current
+		to_chat(S, span_notice("Your infernal nature has allowed you to overcome your clownishness."))
+		S.dna.remove_mutation(CLOWNMUT)
+	switch(demonsin) 
+		if(SIN_GLUTTONY)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/shapeshift/demon/gluttony)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/forcewall/gluttony)
+		if(SIN_GREED)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/shapeshift/demon)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/conjure/summon_greedslots)
+		if(SIN_WRATH)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/shapeshift/demon/wrath)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/inflict_handler/ignite)
+		if(SIN_ENVY)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/shapeshift/demon)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/touch/envy)
+		if(SIN_PRIDE)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/shapeshift/demon)
+			owner.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/conjure/summon_mirror)
+	.=..()
+
+/datum/antagonist/sinfuldemon/on_removal()
+	owner.special_role = null
+	owner.current.faction -= "hell"
+	remove_spells()
+	to_chat(owner.current, span_userdanger("Your infernal link has been severed! You are no longer a demon!"))
+	.=..()
+
+/datum/antagonist/sinfuldemon/proc/remove_spells()
+	for(var/X in owner.spell_list)
+		var/obj/effect/proc_holder/spell/S = X
+		if(is_type_in_typecache(S, demon_spells))
+			owner.RemoveSpell(S)
+
+/datum/antagonist/sinfuldemon/roundend_report()
```

Comment:
no greentext? sad

Reply:
For what? The flavor objectives that autocomplete give you an auto greentext but I haven't tested the assassinate or steal identity ones, though they should also work.